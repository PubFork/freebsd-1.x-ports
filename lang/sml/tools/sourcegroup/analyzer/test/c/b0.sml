signature A = sig
  type t
  structure M :sig val x:t end
end

functor F (A:A) = struct open A end

structure B = F(struct type t=int structure M = struct val x:t=4 end end)
(* B.M *)


functor G (structure D:A) = struct open D end

structure W = struct
  type t=bool
  structure M = struct val x:t=false end
end

structure K = G (structure D = W) 
(* K.M *)

structure L = F (W)
(* L.M *)
