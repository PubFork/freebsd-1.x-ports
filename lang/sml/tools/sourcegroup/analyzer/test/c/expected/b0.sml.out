source sml "c/b0.sml"
  export structure B K L W 
  export signature A 
  export functor F G ;

